src/Tribe/Rewrite.php

```diff
@@ -325,6 +325,46 @@ protected function replace_array_key( &$array, $search, $replace ) {
 		return false;
 	}
 
+	/**
+	 * Makes sure we dont have duplicate canonical variables on the same URL.
+	 *
+	 * @since  TBD
+	 *
+	 * @param  array  $query_vars  Array of the current canonical variables.
+	 * @param  string $url         Which url we are trying to merge canonical variables from.
+	 *
+	 * @return array  Final version correct variable unset and merged.
+	 */
+	protected function merge_canonical_query_vars( array $query_vars, $url = null ) {
+
+		/**
+		 * Creates a map for merging values into one canonical param.
+		 *
+		 * @since TBD
+		 *
+		 * @param array          $map_canonical_query_args Associative array following the format: `[ 'eventDate' => [ 'event-date', 'event_date', 'tribe-bar-date' ], ]`.
+		 * @param string         $url                      The input URL to resolve to a canonical one.
+		 * @param Tribe__Rewrite $this                     This rewrite object.
+		 */
+		$map_canonical_query_args = (array) apply_filters( 'tribe_rewrite_canonical_query_args', [], $url, $this );
+
+		foreach ( $map_canonical_query_args as $canonical_key => $merge_keys ) {
+			foreach( $merge_keys as $merge_key ) {
```

Comment:
:no_entry_sign: There must be a space between foreach and the first parenthesis '('

<sub>hash: 34ca69eedbfc5b745c1342dd95e52002</sub>